Fix: 194 release beschreiben (#246)
it-at-m/lhm_actions: README.md

Reviewer: Will you change the `.releaserc` in a different commit? If yes i think this docs should only be merged if that is done and also it would probably be a good idea to enforce the squash commit title if that is possible, as it currently often happens that that is not done and an additional merge just for the release need to be added (alternative: adding to PR template?)

https://github.com/it-at-m/lhm_actions/blob/8220963279bf17aa53e51eb231acc238eab66632/.releaserc#L5

Reviewer 2: @hupling Pls. synchronize .releaserc and the documentation. Also we should think about changing PR template, so that the developers set the prefixes (e.g. this PR doesn't have a prefix, so semantic versioning will not be used).

Author: also das hat alles meiner Meinung nach nicht funktioniert mit den Titel. Daher muss man es so komisch in die Commit-Message einfügen

Author: <img width="207" height="66" alt="grafik" src="https://github.com/user-attachments/assets/0afc4f2e-69ad-47ba-a46d-1c7eae403e9e" />

wir haben doch hier squash eingestellt.

Author: @simonhir meinte wir sollten nochmal testen, ob ein patch und major überhaupt funktioniert

Author: <img width="1284" height="826" alt="grafik" src="https://github.com/user-attachments/assets/ec0aa3ca-0d40-4851-a287-29009d27bf48" />

also mein Test hat ergeben, dass es trotzdem so geht wie gewollt.

Author: <img width="985" height="601" alt="Image" src="https://github.com/user-attachments/assets/4a16fc37-9808-4956-af3d-2632b1bb3aed" />

habe das jetzt geändert dann reicht es, wenn man im pr titel fix, feat, perf schreibt. bei major muss man dann noch zusätzlich `BREAKING_CHANGE: ` schreiben.

Reviewer 2: @hupling May be a good idea to add this information to the description?

Author: Also ich kann dir sagen, wie es im KPS Projekt war. Ich habe die PR-Description eine Woche lang durchgelesen und dann nie wieder. Einfach Issue Id eingetragen und alles abgehackt. Daher weiß ich jetzt nicht, ob das wirklich einen Mehrwert bringt. Außerdem machen wir sehr selten einen Major-Change

Reviewer 2: @hupling understood that "pull request title and description" is a repo not a pr setting, so it is not necessary to describe.

More about this in the [CODE_OF_CONDUCT](/CODE_OF_CONDUCT.md) file.

## How to do a release?

> We are using [semantic-release](https://github.com/semantic-release/semantic-release) to autogenerate releases. Before merging you need to add in a prefix in the pull request title. [Details](https://github.com/semantic-release/semantic-release#commit-message-format)

Reviewer 2: ```suggestion
We are using [semantic-release](https://github.com/semantic-release/semantic-release) to autogenerate releases. Before merging you need to add a prefix in the pull request title. [Details](https://github.com/semantic-release/semantic-release#commit-message-format)
```

> - `fix:` for patch

Reviewer 2: ```suggestion
- `fix:` for patch versions
```

> - `feat:` for minor

Reviewer 2: ```suggestion
- `feat:` for minor versions
```

> - `perf:` for major additional in the pr description `BREAKING CHANGE:`

Reviewer 2: ```suggestion
- `perf:` for major versions - additional in the pull request description `BREAKING CHANGE:`
```

> - `perf:` for major versions - additional in the pull request description `BREAKING CHANGE:`

Author: ```suggestion
- `perf:` for major versions -  **Note**: When using squash merge, include `BREAKING CHANGE: description of breaking change` in the PR description body, which will be included in the squashed commit message.
```

> - `perf:` for major versions - **Note**: When using squash merge, include `BREAKING CHANGE: description of breaking change` in the PR description body, which will be included in the squashed commit message.

Reviewer: for what does `perf:` stand?

Author: https://github.com/semantic-release/semantic-release#commit-message-format weiß ich nicht. Ich habe das nur von dem Tool übernommen

Reviewer: Ich glaub das ist dann ein Missverständnis `perf` steht für Performance und hat erstmal nichts mit dem Major-Release zu tuen. Ob es Major wird, wird nur über das `BREAKING CHANGE:` gesteuert und der Typ ist egal (würde auch mit `feat` gehen).

Author: tatsächlich. dann wären wir wieder bei dem punkt. lieber die orginal doku lesen, bevor ich etwas falsch interpretiere https://github.com/it-at-m/lhm_actions/pull/246#discussion_r3159138214

Author: https://github.com/semantic-release/commit-analyzer/blob/master/lib/default-release-rules.js aber ich habe jetzt auch nicht getestet, welche Keywörter alles von hier funktionieren

Author: ```suggestion
- `BREAKING CHANGE: ` in the PR description body, which will be included in the squashed commit message for major versions.
```

Author: @simonhir passt das so=

Reviewer: Ja, so wäre auf jeden Fall auch mein Verständnis, wie es funktionieren müsste.

More about this in the [CODE_OF_CONDUCT](/CODE_OF_CONDUCT.md) file.

> # How to make a release?

Author: ```suggestion
## How to make a release?
```

> We are using [semantic-release](https://github.com/semantic-release/semantic-release) to autogenerate releases. Before merging you need to add in a prefix in the commit message in the pull request. 

Author: ```suggestion
We are using [semantic-release](https://github.com/semantic-release/semantic-release) to autogenerate releases. Before merging you need to add in a prefix in the commit message on the pull request. 
```

- fix -> patch
- feat -> minor

> - perf -> major

Reviewer: Is there a reason you not following the conventional commit mapping where breaking changes lead to an major version? By default semantic-release seams to only react to the `BREAKING CHANGE:` in the commit footer, but it would be possible to also react to `!` after the type/scope which is part of the conventional commits spec.

Author: bevor ich jetzt etwas falsch schreibe lösche ich das alles raus und schreibe. Details in der semantic Doku nachlesen
